runtime/vara/src/migrations.rs

```diff
@@ -24,4 +24,94 @@ pub type Migrations = (
     pallet_gear_program::migration::AppendStackEndMigration<Runtime>,
     // migration for removed waiting init list
     pallet_gear::migrations::MigrateWaitingInitList<Runtime>,
+    // substrate v1.3.0
+    pallet_nomination_pools::migration::versioned_migrations::V5toV6<Runtime>,
+    pallet_nomination_pools::migration::versioned_migrations::V6ToV7<Runtime>,
+    staking_v13::MigrateToV13<Runtime>,
 );
+
+mod staking_v13 {
+    use frame_support::{
+        pallet_prelude::{ValueQuery, Weight},
+        storage_alias,
+        traits::{GetStorageVersion, OnRuntimeUpgrade},
+    };
+    use pallet_staking::*;
+    use parity_scale_codec::{Decode, Encode, MaxEncodedLen};
+    use scale_info::TypeInfo;
+    use sp_core::Get;
+    use sp_std::vec::Vec;
+
+    #[cfg(feature = "try-runtime")]
+    use sp_runtime::TryRuntimeError;
+
+    /// Alias to the old storage item used for release versioning. Obsolete since v13.
+    #[storage_alias]
+    type StorageVersion<T: pallet_staking::Config> =
+        StorageValue<Pallet<T>, ObsoleteReleases, ValueQuery>;
+
+    /// Used for release versioning upto v12.
+    ///
+    /// Obsolete from v13. Keeping around to make encoding/decoding of old migration code easier.
+    #[derive(Default, Encode, Decode, Clone, Copy, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
+    enum ObsoleteReleases {
+        V1_0_0Ancient,
+        V2_0_0,
+        V3_0_0,
+        V4_0_0,
+        V5_0_0,  // blockable validators.
+        V6_0_0,  // removal of all storage associated with offchain phragmen.
+        V7_0_0,  // keep track of number of nominators / validators in map
+        V8_0_0,  // populate `VoterList`.
+        V9_0_0,  // inject validators into `VoterList` as well.
+        V10_0_0, // remove `EarliestUnappliedSlash`.
+        V11_0_0, // Move pallet storage prefix, e.g. BagsList -> VoterBagsList
+        V12_0_0, // remove `HistoryDepth`.
+        #[default]
+        V13_0_0, // Force migration from `ObsoleteReleases`.
+    }
+
+    pub struct MigrateToV13<T>(sp_std::marker::PhantomData<T>);
+    impl<T: pallet_staking::Config> OnRuntimeUpgrade for MigrateToV13<T> {
+        #[cfg(feature = "try-runtime")]
+        fn pre_upgrade() -> Result<Vec<u8>, TryRuntimeError> {
+            frame_support::ensure!(
```

Comment:
this will fail on upgrade, shouldn't we just skip here as well?

Reply:
It won't fail. The current StorageVersion is `ObsoleteReleases::V13_0_0`. Check the try-runtime